src/test/ui/generator/auto-trait-regions.rs

```diff
@@ -8,46 +8,48 @@ struct No;
 
 impl !Foo for No {}
 
-struct A<'a, 'b>(&'a mut bool, &'b mut bool, No);
+struct A<'a, 'b: 'a>(&'a mut bool, &'b mut bool, No);
 
-impl<'a, 'b: 'a> Foo for A<'a, 'b> {}
+impl<'a, 'b> Foo for A<'a, 'b> {}
 
 struct OnlyFooIfStaticRef(No);
 impl Foo for &'static OnlyFooIfStaticRef {}
 
 struct OnlyFooIfRef(No);
 impl<'a> Foo for &'a OnlyFooIfRef {}
 
-fn assert_foo<T: Foo>(f: T) {}
+fn assert_foo<T: Foo>(_f: T) {}
 
 fn main() {
-    // Make sure 'static is erased for generator interiors so we can't match it in trait selection
-    let x: &'static _ = &OnlyFooIfStaticRef(No);
+    let z = OnlyFooIfStaticRef(No);
+    let x = &z;
+    //~^ ERROR `z` does not live long enough
     let gen = || {
         let x = x;
         yield;
         assert_foo(x);
     };
-    assert_foo(gen);
-    //~^ ERROR implementation of `Foo` is not general enough
-    //~| ERROR implementation of `Foo` is not general enough
+    assert_foo(gen); // bad
+    drop(z);
+    //~^ ERROR cannot move out of `z` because it is borrowed
 
     // Allow impls which matches any lifetime
-    let x = &OnlyFooIfRef(No);
+    let z = OnlyFooIfRef(No);
+    let x = &z;
     let gen = || {
         let x = x;
         yield;
         assert_foo(x);
     };
     assert_foo(gen); // ok
+    drop(z);
 
-    // Disallow impls which relates lifetimes in the generator interior
+    let x = &mut true;
```

Comment:
I changed this test because I'm not sure what to make of it, and it wasn't compiling anyways..